hazelcast/src/main/java/com/hazelcast/internal/config/YamlConfigSchemaValidator.java

```diff
@@ -70,15 +77,16 @@ public void validate(YamlMapping rootNode) {
         try {
             // this could be expressed in the schema as well, but that would make all the schema validation errors much harder
             // to read, so it is better to implement it here as a semantic check
-            long definedRootNodeCount = PERMITTED_ROOT_NODES.stream()
+            List<String> definedRootNodes = PERMITTED_ROOT_NODES.stream()
                     .filter(rootNodeName -> rootNode != null && rootNode.child(rootNodeName) != null)
-                    .count();
-
-            if (definedRootNodeCount != 1) {
+                    .collect(toList());
+            if (definedRootNodes.size() != 1) {
                 throw new SchemaViolationConfigurationException(
                         "exactly one of [hazelcast], [hazelcast-client] and [hazelcast-client-failover] should be present in the"
-                                + " root schema document, " + definedRootNodeCount + " are present",
+                                + " root schema document, " + definedRootNodes.size() + " are present",
                         "#", "#", emptyList());
+            } else if (!"false".equals(ROOT_LEVEL_INDENTATION_CHECK_ENABLED.getSystemProperty())) {
```

Comment:
You can use `new HazelcastProperties(System.getProperties()).getBoolean(ROOT_LEVEL_INDENTATION_CHECK_ENABLED)` here and in `AbstractConfigBuilder#shouldValidateTheSchema` as well.

Reply:
Done at both places.